Objective-C/Internal/CBLFleece.hh

```diff
@@ -56,15 +56,16 @@ namespace cbl {
     class DocContext : public fleece::MContext {
     public:
         DocContext(CBLDatabase *db, CBLC4Document* __nullable doc);
+        DocContext(); // setJSON uses this context, without db or c4doc
         
-        CBLDatabase* database() const   {return _db;}
+        CBLDatabase* __nullable database() const   {return _db;}
```

Comment:
DocContext requires a database object for locking. I don't remember all of the detail as it has been a long time. Can you dig this up to see if it's OK not to have the database?

Reply:
It was required for creating the blob(1).  But we are already creating Blob without DB(2), since the documents created from SetJSON lacks DB. 

(1) https://github.com/couchbase/couchbase-lite-ios/blob/7d9724464d99e5b5cbb3203a58287f29c4f8ef0b/Objective-C/Internal/CBLFleece.mm#L66

(2) https://github.com/couchbase/couchbase-lite-ios/blob/7d9724464d99e5b5cbb3203a58287f29c4f8ef0b/Objective-C/Internal/CBLData.mm#L57